newdle/client/src/components/AvailabilityRing.js

```diff
@@ -0,0 +1,76 @@
+import React from 'react';
+
+AvailabilityRing.defaultProps = {
+  strokeWidth: 10,
+  radius: 50,
+};
+
+export default function AvailabilityRing({
+  available,
+  ifNeeded,
+  totalParticipants,
+  strokeWidth,
+  radius,
+}) {
+  let size = radius * 1.1;
+  let stroke = 7;
+  let normalizedRadius = radius - stroke * 2;
+  let circumference = normalizedRadius * 2 * Math.PI;
+
+  const greenOffset = `rotate(-90 ${size} ${size})`;
+  const yellowOffset = `rotate(${-90 + (available / totalParticipants) * 360} ${size} ${size})`;
+  const redOffset = `rotate(${-90 +
+    ((available + ifNeeded) / totalParticipants) * 360} ${size} ${size})`;
+
+  let style = {
+    'width': '25%',
+    'color': 'black',
+    'text-align': 'center',
```

Comment:
This should be `textAlign` (since the style dict in React uses the JS names for style attributes, not the CSS ones). You can also get rid of the quotes around the keys now since everything is a valid JS identifier.